step-functions-data-science-sdk/automate_model_retraining_workflow/automate_model_retraining_workflow.ipynb

```diff
@@ -0,0 +1,983 @@
+    "* [**FailStateStep**](https://aws-step-functions-data-science-sdk.readthedocs.io/en/latest/states.html#stepfunctions.steps.states.Fail) - Creates fail state step within our workflow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# SageMaker expects unique names for each job, model and endpoint. \n",
+    "# If these names are not unique the execution will fail.\n",
+    "execution_input = ExecutionInput(schema={\n",
+    "    'TrainingJobName': str,\n",
+    "    'GlueJobName': str,\n",
+    "    'ModelName': str,\n",
+    "    'EndpointName': str,\n",
+    "    'LambdaFunctionName': str\n",
+    "})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create an ETL step with AWS Glue\n",
+    "In the following cell, we create a Glue step thats runs an AWS Glue job. The Glue job extracts the latest data from our source database, removes unnecessary columns, splits the data in to training and validation sets, and saves the data to CSV format in S3. Glue is performing this extraction, transformation, and load (ETL) in a serverless fashion, so there are no compute resources to configure and manage. See the [GlueStartJobRunStep](https://aws-step-functions-data-science-sdk.readthedocs.io/en/latest/compute.html#stepfunctions.steps.compute.GlueStartJobRunStep) Compute step in the AWS Step Functions Data Science SDK documentation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "etl_step = steps.GlueStartJobRunStep(\n",
+    "    'Extract, Transform, Load',\n",
+    "    parameters={\"JobName\": execution_input['GlueJobName'],\n",
+    "                \"Arguments\":{\n",
+    "                    '--S3_SOURCE': data_source,\n",
+    "                    '--S3_DEST': 's3a://{}/{}/'.format(bucket, project_name),\n",
+    "                    '--TRAIN_KEY': train_prefix + '/',\n",
+    "                    '--VAL_KEY': val_prefix +'/'}\n",
+    "               }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a SageMaker Training Step \n",
+    "\n",
+    "In the following cell, we create the training step and pass the estimator we defined above. See  [TrainingStep](https://aws-step-functions-data-science-sdk.readthedocs.io/en/latest/sagemaker.html#stepfunctions.steps.sagemaker.TrainingStep) in the AWS Step Functions Data Science SDK documentation to learn more."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training_step = steps.TrainingStep(\n",
+    "    'Model Training', \n",
+    "    estimator=xgb,\n",
+    "    data={\n",
+    "        'train': sagemaker.s3_input(train_data, content_type='csv'),\n",
```

Comment:
[Minor] Same comments as the other ones. Consider importing this to avoid repeating "sagemaker."

Reply:
Good idea. Resolved in the most recent commit.